lib/vanagon/component.rb

```diff
@@ -265,7 +265,8 @@ def fetch_url(options)
     # a rewritten URL if there's any value for #url.
     def deprecated_rewrite_url
       return nil unless url
-      Vanagon::Component::Source::Rewrite.parse_and_rewrite(url)
+      rewritten_url = Vanagon::Component::Source::Rewrite.parse_and_rewrite(url)
+      url == rewritten_url ? nil : rewritten_url
```

Comment:
Is there a non-deprecated method to fix this in?

Reply:
This is the method that's called directly when `mirrors` is trying to populate its default value. The other option is to do the check in the `mirrors` accessor if this just returned `url`

Reply:
Maybe this isn't the place for this work, but I feel like `mirrors` shouldn't be calling a deprecated method. Like, why not address this issue in `Vanagon::Component::Source::Rewrite.parse_and_rewrite` and have `mirrors` use that?

Reply:
The entire rewrite system is deprecated. This method could perhaps better be called `call_deprecated_rewrite_for_compatibility`, but that's a very long method name

Reply:
![](https://media.giphy.com/media/l4FGvWSazaXiqVDRS/giphy.gif)

Reply:
![](https://media.giphy.com/media/l4FGvWSazaXiqVDRS/giphy.gif)

Reply:
![](https://media.giphy.com/media/l0Iy1rwL0a15MsSXu/giphy.gif)